Vet clinic database: query and update animals table

JuanDiegoGuerra opened the pull request:
### Activity Requirements - Vet Clinic ( query and update animals table )

- Add a column `species` .
- Inside a transaction update the `animals` table by setting the `species` column to `unspecified`. Verify that change was made. Then roll back the change and verify that the `species` columns went back to the state before the transaction.
- Inside a transaction:
   - Update the animals table by setting the `species` column to `digimon` for all animals that have a name ending in mon.
   - Update the animals table by setting the `species` column to `pokemon` for all animals that don't have species already set.
   - Verify that changes were made.
   - Commit the transaction.
   - Verify that changes persist after commit.
![image](https://github.com/JuanDiegoGuerra/Vet-Clinic_Database/assets/45726631/38676f21-9cbd-4dca-bef0-1e0aa34d12ed)
- Inside a transaction:
  - Delete all animals born after Jan 1st, 2022.
  - Create a savepoint for the transaction.
  - Update all animals' weight to be their weight multiplied by -1.
  - Rollback to the savepoint
  - Update all animals' weights that are negative to be their weight multiplied by -1.
  - Commit transaction
![image](https://github.com/JuanDiegoGuerra/Vet-Clinic_Database/assets/45726631/76118860-951b-4383-a4ba-8e9dbee31b9e)
---
- Queries : 
   - How many animals are there?
![image](https://github.com/JuanDiegoGuerra/Vet-Clinic_Database/assets/45726631/284be77c-25e5-4e7f-b1c0-9ccf15f73f8d)
   - How many animals have never tried to escape?
![image](https://github.com/JuanDiegoGuerra/Vet-Clinic_Database/assets/45726631/26bbd9ea-b2c6-4553-84fe-421eb4f5a5a5)
   - What is the average weight of animals?
![image](https://github.com/JuanDiegoGuerra/Vet-Clinic_Database/assets/45726631/e79952fc-5589-4015-a1c6-44452171bace)
   - Who escapes the most, neutered or not neutered animals?
![image](https://github.com/JuanDiegoGuerra/Vet-Clinic_Database/assets/45726631/0cc08e25-447f-4d61-a976-96f3beb3ff38)
   - What is the minimum and maximum weight of each type of animal?
![image](https://github.com/JuanDiegoGuerra/Vet-Clinic_Database/assets/45726631/a55c9a42-a5ca-4d69-9b28-f1aff27773e5)
   - What is the average number of escape attempts per animal type of those born between 1990 and 2000?
![image](https://github.com/JuanDiegoGuerra/Vet-Clinic_Database/assets/45726631/98d248e9-209f-4885-a138-f504b9413a89)

JuanDiegoGuerra committed:
Update schema.sql

Alter table to add a column

JuanDiegoGuerra committed:
Update data.sql

INSERT VALUES INTO animals' table

JuanDiegoGuerra committed:
Update queries.sql

All the queries for project requirements

JuanDiegoGuerra committed:
Create LICENSE

JuanDiegoGuerra committed:
Update queries.sql

joskalenda commented on queries.sql:
```diff
@@ -23,3 +24,64 @@ SELECT * FROM animals WHERE name NOT IN ('Gabumon');
+WHERE name LIKE '%mon';
+
+/*Update the animals table by setting the species column to pokemon for all animals that don't have species already set.*/
+UPDATE animals
+SET species = 'pokemon'
+WHERE name NOT LIKE '%mon';
```
- [ ] Dear @JuanDiegoGuerra, am sure this query interprets the requirement, you are just updating any species to Pokemon where the name is anything not similar to mon, but the requirement says to update the animals table by setting the species column to Pokemon for all animals that `don't have species already set` I think it refers to a `null` value 
<img width="957" alt="1" src="https://github.com/JuanDiegoGuerra/Vet-Clinic_Database/assets/77843050/d2510ed1-d541-4b18-ad79-f47ca290b9bd">

joskalenda reviewed (changes requested):
| STATUS: REQUIRED CHANGES ♻️  |
|--------------------------------|
| Hi @JuanDiegoGuerra , |
|Good job so far! There are some issues that you still need to work on to go to the next project but you are almost there! |
| <img src="https://media3.giphy.com/media/UdaSUYClr92cjWpeVx/giphy.gif"/> |

## Highlights

- [ ]... ✅ Clean code
- [ ].. ✅ Good readme

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

-------
> Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me(@joskalenda) in your question so I can receive the notification.

> **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

JuanDiegoGuerra committed:
Update queries.sql

Update the animals table by setting the species column to pokemon for all animals that don't have species already set.

JuanDiegoGuerra commented:
###  Required changes made 🛠

- **Fixed:** _"Dear JuanDiegoGuerra, am sure this query interprets the requirement, you are just updating any species to Pokemon where the name is anything not similar to mon, but the requirement says to update the animals table by setting the species column to Pokemon for all animals that don't have species already set I think it refers to a null value"_ ✅

![todolist1](https://user-images.githubusercontent.com/45726631/231703283-083f07c2-f037-4fa4-a834-be7d3bd26e06.gif)

t-yanick reviewed (approved):
# STATUS: MILESTONE APPROVED :1st_place_medal: 

Hi @JuanDiegoGuerra , 

### Great job !!!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  Required changed implemented 👍 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:  
Congratulations! 🎉 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
